sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java

```diff
@@ -0,0 +1,122 @@
+import com.azure.resourcemanager.servicebus.models.Topic;
+import com.azure.resourcemanager.servicebus.models.Topics;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import reactor.util.function.Tuple3;
+import reactor.util.function.Tuples;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ServiceBusTopicSubscriptionCrudTests extends AbstractResourceCrudTests<ServiceBusSubscription,
+    Tuple3<String, String, String>> {
+
+    private static final String NAMESPACE = "namespace";
+    private static final String TOPIC_NAME = "topic";
+    private static final String SUBSCRIPTION_NAME = "subscription";
+
+    @Override
+    AbstractResourceCrud<ServiceBusSubscription, Tuple3<String, String, String>> getResourceCrud() {
+        return new ServiceBusTopicSubscriptionCrud(resourceManager, resourceMetadata);
+    }
+
+    @Override
+    void getStubManagementException(int statusCode, String message) {
+        ServiceBusNamespaces namespaces = mock(ServiceBusNamespaces.class);
+        ServiceBusNamespace namespace = mock(ServiceBusNamespace.class);
+
+        when(resourceManager.serviceBusNamespaces()).thenReturn(namespaces);
+        ManagementException exception = getManagementException(statusCode, message);
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), getKey().getT1()))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        Topic topic = mock(Topic.class);
+        ServiceBusSubscriptions serviceBusSubscriptions = mock(ServiceBusSubscriptions.class);
+        when(namespace.topics()).thenReturn(topics);
+        when(topics.getByName(getKey().getT2())).thenReturn(topic);
+        when(topic.subscriptions()).thenReturn(serviceBusSubscriptions);
+        when(serviceBusSubscriptions.getByName(getKey().getT2())).thenThrow(exception);
+    }
+
+    @Override
+    void createStubManagementException() {
+        ServiceBusNamespaces namespaces = mock(ServiceBusNamespaces.class);
+        ServiceBusNamespace namespace = mock(ServiceBusNamespace.class);
+        when(resourceManager.serviceBusNamespaces()).thenReturn(namespaces);
+        ManagementException exception = getManagementException(500, "Create service bus namespace exception");
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), getKey().getT1()))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        Topic topic = mock(Topic.class);
+        ServiceBusSubscriptions serviceBusSubscriptions = mock(ServiceBusSubscriptions.class);
+        when(namespace.topics()).thenReturn(topics);
+        when(topics.getByName(getKey().getT2())).thenReturn(topic);
+        when(topic.subscriptions()).thenReturn(serviceBusSubscriptions);
+        when(serviceBusSubscriptions.getByName(getKey().getT3())).thenThrow(exception);
```

Comment:
Seems like we can remove the namespaces in this case?

Try to minimize the unnecessary code.